core/src/test/java/com/datastrato/gravitino/storage/relational/TestRelationalEntityStore.java

```diff
@@ -5,100 +5,296 @@
+  private static EntityStore entityStore = null;
 
   @BeforeAll
   public static void setUp() {
+    File dir = new File(DB_DIR);
+    if (dir.exists() || !dir.isDirectory()) {
+      dir.delete();
+    }
+    dir.mkdirs();
+
+    // Use H2 DATABASE to simulate MySQL
     Config config = Mockito.mock(Config.class);
-    Mockito.when(config.get(Configs.ENTITY_STORE)).thenReturn(Configs.RELATIONAL_ENTITY_STORE);
-    Mockito.when(config.get(Configs.ENTITY_RELATIONAL_STORE))
-        .thenReturn(DEFAULT_ENTITY_RELATIONAL_STORE);
-    store = EntityStoreFactory.createEntityStore(config);
-    store.initialize(config);
-    Assertions.assertTrue(store instanceof RelationalEntityStore);
+    Mockito.when(config.get(ENTITY_STORE)).thenReturn(RELATIONAL_ENTITY_STORE);
+    Mockito.when(config.get(ENTITY_RELATIONAL_STORE)).thenReturn(DEFAULT_ENTITY_RELATIONAL_STORE);
+    Mockito.when(config.get(ENTITY_RELATIONAL_JDBC_BACKEND_DB_TYPE))
+        .thenReturn(DEFAULT_ENTITY_RELATIONAL_JDBC_BACKEND_DB_TYPE);
+    Mockito.when(config.get(ENTITY_RELATIONAL_JDBC_BACKEND_URL))
+        .thenReturn(String.format("jdbc:h2:%s;DB_CLOSE_DELAY=-1;MODE=MYSQL", DB_DIR));
+    Mockito.when(config.get(ENTITY_RELATIONAL_JDBC_BACKEND_USER)).thenReturn("root");
+    Mockito.when(config.get(ENTITY_RELATIONAL_JDBC_BACKEND_PASSWORD)).thenReturn("123");
+    Mockito.when(config.get(ENTITY_RELATIONAL_JDBC_BACKEND_DRIVER)).thenReturn("org.h2.Driver");
+    entityStore = EntityStoreFactory.createEntityStore(config);
+    entityStore.initialize(config);
+
+    // Read the ddl sql to create table
+    String scriptPath = "h2/h2-init.sql";
+    try (SqlSession sqlSession =
+            SqlSessionFactoryHelper.getInstance().getSqlSessionFactory().openSession(true);
+        Connection connection = sqlSession.getConnection();
+        Statement statement = connection.createStatement()) {
+      URL scriptUrl = ClassLoader.getSystemResource(scriptPath);
+      if (scriptUrl == null) {
+        throw new IllegalStateException("Cannot find init sql script:" + scriptPath);
+      }
+      StringBuilder ddlBuilder = new StringBuilder();
+      try (InputStreamReader inputStreamReader =
+              new InputStreamReader(
+                  Files.newInputStream(Paths.get(scriptUrl.getPath())), StandardCharsets.UTF_8);
+          BufferedReader bufferedReader = new BufferedReader(inputStreamReader)) {
+        String line;
+        while ((line = bufferedReader.readLine()) != null) {
+          ddlBuilder.append(line).append("\n");
+        }
+      }
+      statement.execute(ddlBuilder.toString());
+    } catch (Exception e) {
+      throw new IllegalStateException("Create tables failed", e);
+    }
+  }
+
+  @AfterEach
+  public void destroy() {
+    truncateAllTables();
   }
 
   @AfterAll
-  public static void teardown() throws IOException {
-    store.close();
-    store = null;
+  public static void tearDown() throws IOException {
```

Comment:
I think we can catch the `Exception` and log for it. Because this will not impact the result of UTs.

Reply:
Done